### ISSUE TITLE
Bad padding-left on header bar user-menu

### ISSUE THREAD

**Issue by milouse** [diaspora/diaspora#8115]
I have a weird display bug since a long time on my instance. As no-one has already reported it, it may be a problem only on my instance, but maybe no one really care.

All the user-menu items have a padding-left of 55px in desktop mode. The cause line is [this one](https://github.com/diaspora/diaspora/blob/b94ef57bbccd1d007c2e51cee627e823677c36e4/app/assets/stylesheets/header.scss#L237). If I remove it, everything come back to normal.

![Capture du 2020-03-23 10-55-21](https://user-images.githubusercontent.com/349239/77304485-cf106680-6cf4-11ea-9df3-cb2b6b406ad2.png)

Does anyone has the same problem? What is the reason behind this big padding?

Thank you very much

**Comment by Flaburgan** [comment]
Hi! This isn't a bug, it is the expected behavior, so the menu items are aligned with your username. But that can be changed if we (you included) find it better to not have that left padding.

**Comment by milouse** [comment]
Maybe it's the french translation of logout (« Se déconnecter ») and admin (« Administration »), but as you see in my screenshot, the result is a bit ugly (for me, at least) as the word are very close to right side of the menu. I was even wondering if my menu was lacking icons or something else.

I understand your point, but maybe we should enlarge the menu in order to give more room to the right of the words? But my opinion is that removing the padding to make text align with the avatar would be better.

**Comment by SuperTux88** [comment]
It's probably a mix between a short name (the dropdown is as wide as your name) and long french translations. But I agree that in this edge case, it's to close to the right border of the dropdown, so we should probably add a little padding at the right side to give it a little more space.

**Comment by milouse** [comment]
Do you have a screenshot of the actual and expected result (on your instances for exemple?)

**Comment by SuperTux88** [comment]
On my user it looks like this:
![image](https://user-images.githubusercontent.com/458548/77455777-965ab500-6dfa-11ea-8041-b64270304a34.png)

But if you change your name to something longer, or switch to a language with shorter translations, it would look the same on your instance, so it's not about the instance, it's about the individual users.

And if we enforce padding to the right, that wouldn't change my result, but it wouldn't be as close to the right border on your view anymore.

**Comment by milouse** [comment]
Yes I already understood that it was not because of my instance… but I would like to see another wild exemple (like other system fonts…). Sorry for bothering you :/

I think I maybe find a good compromise. I add the following rules to my stylesheet. What do you think of it?

```css
@media (min-width: 992px) {
    #user-menu span.user-name {
        display: inline-block;
        min-width: 120px;
    }
}
```

That looks like, with short names:

![Capture du 2020-03-24 18-43-17](https://user-images.githubusercontent.com/349239/77459197-a1641400-6dff-11ea-9799-9b8d80988f73.png)
![Capture du 2020-03-24 18-43-40](https://user-images.githubusercontent.com/349239/77459203-a32dd780-6dff-11ea-9bf1-188f022d6e71.png)

And with long names (`min-width` is ignored, thus obviously that's the default rendering):

![Capture du 2020-03-24 18-44-52](https://user-images.githubusercontent.com/349239/77459205-a45f0480-6dff-11ea-8ac2-d32f30cb040d.png)
![Capture du 2020-03-24 18-45-10](https://user-images.githubusercontent.com/349239/77459207-a5903180-6dff-11ea-865c-cfdc9ba29ccf.png)

**Comment by Flaburgan** [comment]
Just to understand correctly, what's bothering you, the missing space on the right, or the wasted space on the left? 

Btw, I remember a discussion about adding icons in the left space for each menu item. I don't remember if we took any decision though.

**Comment by milouse** [comment]
> Just to understand correctly, what's bothering you, the missing space on the right, or the wasted space on the left?

Both actually. But from your previous answer, I understand that you don't want to remove the left space, hence my proposal about enforcing a confortable right space in any cases.

As stated before, my own opinion is that left space should be removed to align text on the avatar, not the name. But I understand that decision was already taken and I don't want to modify them.

**Comment by Flaburgan** [comment]
I don't have a strong opinion here honestly, I'm also fine with the left padding removed.

**Comment by Flaburgan** [comment]
As I'm dealing with #7676 and adding a `max-width` for long usernames I was wondering if I should add a `min-width` at the same time to deal with this. However I'm not too sure it's better.

Here is how diaspora* actually looks like:
![Capture d’écran de 2021-10-30 11-45-09](https://user-images.githubusercontent.com/930064/139528373-f5a2e657-0de5-4443-bf5a-81f4e8cdde09.png)
![Capture d’écran de 2021-10-30 11-55-19](https://user-images.githubusercontent.com/930064/139528400-ef459847-eef2-4dd7-b12d-f5cc51087c65.png)

Here is how it could looks like with a min width:
![Capture d’écran de 2021-10-30 11-57-06](https://user-images.githubusercontent.com/930064/139528444-c16c9857-78de-4916-8a25-f20d5d2102f2.png)
![Capture d’écran de 2021-10-30 11-56-55](https://user-images.githubusercontent.com/930064/139528450-b19258c7-752b-4031-8465-df645eca8ccd.png)

So when the menu is open, it looks better for sure. But when it isn't, the caret looks lost in the right of the screen. What do you think?

**Comment by goobertron** [comment]
I have two thoughts on this that might help 'fix' it.

1. Create the div that contains the drop-down menu based on the field that contains the name only, not the avatar.
2. Place the avatar to the right of the user name in the header bar, aligned left of the down arrow.

Both of these, if I'm right, would mean the menu items would be at the left-hand side of the drop-down area, with no indent needed. The second would also mean that the down arrow is not isolated if the user name is very short (mentioned in @Flaburgan's latest comment).

Any use?

**Comment by SuperTux88** [comment]
> I was wondering if I should add a min-width at the same time to deal with this.

We already have a `min-width`:

https://github.com/diaspora/diaspora/blob/d1e0b163e25cca972f8a382b7206022f3701bb90/app/assets/stylesheets/header.scss#L222-L228

But it's too small for some translations (french for example), where it then comes to close to the right with longer texts.

And as already mentioned, I don't like to just increase that `min-width`, I'd rather have a padding so the width depends on the translation. There already is a padding for menu entry links, but they are ignored and the text goes behind the limit and the menu doesn't grow with longer text:

![image](https://user-images.githubusercontent.com/458548/139556749-88e5c611-5048-4b48-808c-d66b69bb0a5c.png)

**Comment by Flaburgan** [comment]
Another solution could be to move the avatar to the right:
![image](https://user-images.githubusercontent.com/930064/199009137-cc5b46d9-a7c2-455c-ac80-aee0ce9bb44c.png)

But that doesn't really work when it's closed:
![image](https://user-images.githubusercontent.com/930064/199009174-3e9eb03e-3649-4556-8e46-2290ea0c1a43.png)
